src/e2e-test/regression-test/util/generate-ground-truths.js

```diff
@@ -0,0 +1,100 @@
+
+const generateTruths = (workerPool, basePath, domain, subCases) => {
+    const templateBasePath = path.join(constants.TEMPLATE_FILES_LOCATION, domain);
+    const dataBasePath = path.join(constants.SAMPLE_DATA_LOCATION, domain);
+    const subPath = path.join(basePath, domain);
+
+    return subCases.map(subCase => new Promise((fulfill, reject) => {
+        const templateContent = fs.readFileSync(path.join(templateBasePath, subCase.templateFile));
+        const dataContent = fs.readFileSync(path.join(dataBasePath, subCase.dataFile));
+        const subTemplatePath = path.join(subPath, subCase.templateFile);
+        
+        if (!fs.existsSync(subTemplatePath)) {
+            fs.mkdirSync(subTemplatePath, { recursive: true });
+        }
+
+        const payload = {
+            type: `/api/convert/:srcDataType`,
+            srcDataType: domain,
+            templateBase64: Buffer.from(templateContent).toString('base64'),
+            srcDataBase64: Buffer.from(dataContent).toString('base64')
+        };
+
+        workerPool.exec(payload)
+            .then(result => {
+                const filePath = path.join(subTemplatePath, utils.getGroundTruthFileName(subCase));
+                fs.writeFile(filePath, JSON.stringify(result.resultMsg, null, 4), 'UTF8', error => {
+                    if (error) {
+                        return reject(error);
+                    }
+                    fulfill(filePath);
+                });
+            })
+            .catch(reject);
+    }));
+};
+
+const main = basePath => {
+    const prompt = `
+        The truths files are already exist in ${basePath}. 
+        Please remove them manually for the normal operation of the program.
+    `;
+    
+    return new Promise((fulfill, reject) => {
+        truthsExist(basePath)
+            .then(flag => {
+                if (flag) {
+                    fulfill(prompt);
+                }
+                const workerPath = path.join(__dirname, '../../../lib/workers/worker.js');
+                const workerPool = new WorkerPool(workerPath, require('os').cpus().length);
+                const cdaPromises = generateTruths(workerPool, basePath, 'cda', cases.cdaCases);
+                const hl7v2Promises = generateTruths(workerPool, basePath, 'hl7v2', cases.hl7v2Cases);
+        
+                const cdaFinalPromise = Promise.all(cdaPromises);
+                const hl7v2FinalPromise = Promise.all(hl7v2Promises);
+        
+                Promise.all([ cdaFinalPromise, hl7v2FinalPromise ])
+                    .then(fulfill)
+                    .catch(reject);
+            })
+            .catch(reject);
+    });
+};
+
+// Be very careful before running this file and make sure that you indeed want
+// to generate new ground truths.
+// command: `node --experimental-worker .\src\regr-test\util\gen-ground-truths.js`
+// Before running, uncomment the following lines and comment them in time after running.
+
+// main(path.join(__dirname, '../data'))
```

Comment:
can we simply uncomment this code? it seems odd here to comment/uncomment manually

Reply:
If the code is uncommented, every time the module is referenced, the main function is executed once, which may not be the behavior we expect. I do agree it's not elegant to manually control the behaviour, but it may be a little complex if we want to generate ground truths in command line without the commented lines.

Reply:
Maybe we can move the cmd script out from this module?

Reply:
Fixed.